src/main/java/ch/njol/skript/conditions/CondMatches.java

```diff
@@ -0,0 +1,82 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+
+package ch.njol.skript.conditions;
+
+import java.util.Arrays;
+import java.util.regex.Pattern;
+
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.util.Kleenean;
+
+@Name("Matches")
+@Description("Test whether strings match defined regular expressions")
+@Examples({"on chat:",
+	"\tif message matches \"\\d\":",
+	"\t\tsend \"Message contains a digit!\""})
+@Since("INSERT VERSION")
+public class CondMatches extends Condition {
+	
+	static {
+		Skript.registerCondition(CondMatches.class, "%strings% match[es] %strings%");
```

Comment:
I would make using `matches` required. Using just `match` sound weird to me. Are there any situations where using `match` would be better though?

Reply:
Yeah, in %strings% match %string%

Reply:
Even with single strings, you should still use matches, no?
See this example:
`
if "123456 HELLO" match "\d":
vs
if "123456 HELLO" matches "\d":
`

Reply:
Yeah but %strings% means multiple ones
"hello" and "hey" match "\d+"

Reply:
Woops! I read your original message as `%string% match %string%`, not `%strings% match %string%`

Sorry lol